homework_16_240123/src/MyArrayList.java

```diff
@@ -0,0 +1,177 @@
+        }
+        System.arraycopy(arrayElements, index, arrayElements, index + 1, size - index);
+        arrayElements[index] = element;
+        size++;
+    }
+
+    /**
+     * Retrieves an element by its position index. In case provided index in out of the list bounds it
+     *
+     * @param index index of the element
+     * @return an element
+     * @throws IndexOutOfBoundsException
+     */
+    public T get(int index) {
+        Objects.checkIndex(index, size);
+        return (T) arrayElements[index];
+    }
+
+    /**
+     * Returns the first element of the list. Operation is performed in constant time 0 (1)
+     *
+     * @return the first element of the list
+     * @throws java.util.NoSuchElementException if list is empty
+     */
+    public T getFirst() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return get(0);
+    }
+
+    /**
+     * Returns the last element of the list. Operation is performed in constant time 0 (1)
+     *
+     * @return the last element of the list
+     * @throws java.util.NoSuchElementException if list is empty
+     */
+    public T getLast() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return get(size - 1);
+    }
+
+    /**
+     * Changes the value of array at specific position. In case provided index in out of the list bounds it
+     *
+     * @param index   position of value
+     * @param element a new value
+     * @throws IndexOutOfBoundsException
+     */
+    public void set(int index, T element) {
+        Objects.checkIndex(index, size);
+        arrayElements[index] = element;
+    }
+
+    /**
+     * Removes an elements by its position index. In case provided index in out of the list bounds it
+     *
+     * @param index element index
+     * @return deleted element
+     * @throws IndexOutOfBoundsException
+     */
+    public T remove(int index) {
+        Objects.checkIndex(index, size);
+        T removedElement = (T) arrayElements[index];
+        System.arraycopy(arrayElements, index + 1, arrayElements, index, size - index - 1);
+        size--;
+        return removedElement;
+    }
+
+    /**
+     * Checks for existing of a specific element in the list
+     *
+     * @param element is element
+     * @return If element exist method returns true, otherwise it returns false
+     */
+    public boolean contains(T element) {
+        for (int i = 0; i < size; i++) {
+            if (element.equals(arrayElements[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a list is empty
+     *
+     * @return {@code true} if list is empty, {@code false} otherwise
+     */
+    public boolean isEmpty() {
+        if (size == 0) {
+            return true;
+        } else {
+            return false;
+        }
+//        return size == 0; // or so it is possible in one line
```

Comment:
size == 0 is also correct and more elegant

Reply:
Fixed.